### PR TITLE
Fix memory leak caused by libupower

### PR DESCRIPTION
**Pull request by uklotzde**
A simple fix to prevent a potential memory leak:
https://gitlab.freedesktop.org/upower/upower/issues/14

### CODE REVIEW

File: src/util/battery/batterylinux.cpp [hunk 1]
```diff
@@ -31,7 +31,12 @@ void BatteryLinux::read() {
     up_client_enumerate_devices_sync(client, NULL, NULL);
 #endif
 
+#if UP_CHECK_VERSION(0, 99, 8)
+    GPtrArray* devices = up_client_get_devices2(client);
+#else
+    // See also: https://gitlab.freedesktop.org/upower/upower/issues/14
```
Comment (daschuer) [inline src/util/battery/batterylinux.cpp]: I think it is better to drop some words about the differences here. One time the free function is set and one time not.

**Review by daschuer — CHANGES_REQUESTED** [review]
Did we do the mentioned mistake from the linked issue when calling
g_ptr_array_free(devices, TRUE);
blow?
Do we need to special case the code there as well?
From our code it looks like both get devices functions are 1:1 exchangeable. But in this case there would be no reason to introduce a second version cluttering the public API.

**Comment by daschuer** [comment]
Thanks for taking cate 😀

**Comment by uklotzde** [comment]
Indeed, with the old version we actually have a memory leak.

**Comment by uklotzde** [comment]
Good catch. I was just too lazy to dig deeper. Memory leak should be fixed now by borrowing a line of code from the patch.

**Review by daschuer — APPROVED** [review]
LGTM, Thank you.

**Pull request merged**